File: ios/LoverApp/Features/Chat/MeetupViews.swift
import SwiftUI

// v1.6.1 Feature 6 — meet-up UI: countdown banner, set-date sheet, and the
// "take a selfie" prompt shown when the meet-up day arrives.

/// Countdown pill shown at the top of the chat. Tapping it opens the set sheet
/// (to change/clear) or, when nothing is set, prompts to set one.
struct MeetupBanner: View {
    @Environment(\.theme) private var theme
    let upcoming: Meetup?
    let onTap: () -> Void

    var body: some View {
        Button(action: onTap) {
            HStack(spacing: 8) {
                Text("💕")
                if let m = upcoming {
                    let d = m.daysUntil
                    Text(d <= 0 ? "今日見面！" : "仲有 \(d) 日見面")
                        .font(DSText.ui(theme, 13, weight: .semibold))
                        .foregroundStyle(theme.ink)
                    Text("· \(m.title)")
                        .font(DSText.mono(theme, 11))
                        .foregroundStyle(theme.inkMuted)
                        .lineLimit(1)
                } else {
                    Text("設定下次見面")
                        .font(DSText.ui(theme, 13, weight: .semibold))
                        .foregroundStyle(theme.rose)
                }
                Spacer()
                Image(systemName: "chevron.right")
                    .font(.system(size: 11, weight: .semibold))
                    .foregroundStyle(theme.inkMuted)
            }
            .padding(.horizontal, 14)
            .padding(.vertical, 8)
            .background(theme.roseSoft)
            .overlay(Rectangle().frame(height: 0.5).foregroundStyle(theme.line),
                     alignment: .bottom)
        }
        .buttonStyle(.plain)
    }
}

/// Date + title picker to create the next meet-up.
struct SetMeetupSheet: View {
    @Environment(\.theme) private var theme
    let existing: Meetup?
    let onCreate: (_ dateISO: String, _ title: String) -> Void
    let onCancel: (_ existingId: UUID?) -> Void   // cancel an existing meet-up
    let onClose: () -> Void

    @State private var date = Date()
    @State private var title = ""

    var body: some View {
        ZStack {
            Color.black.opacity(0.4).ignoresSafeArea().onTapGesture { onClose() }
            VStack(spacing: 16) {
                Text("下次幾時見面？")
                    .font(DSText.ui(theme, 17, weight: .semibold))
                    .foregroundStyle(theme.ink)

                DatePicker("", selection: $date, in: Date()..., displayedComponents: .date)
                    .datePickerStyle(.graphical)
                    .tint(theme.rose)

                TextField("做咩？(例如：睇戲、食飯)", text: $title)
                    .font(DSText.ui(theme, 14))
                    .padding(12)
                    .background(theme.paperAlt, in: RoundedRectangle(cornerRadius: 12))

                Button {
                    onCreate(LocalDate.string(from: date), title)
                    onClose()
                } label: {
                    Text("確定")
                        .font(DSText.ui(theme, 15, weight: .semibold))
                        .foregroundStyle(.white)
                        .frame(maxWidth: .infinity)
                        .padding(.vertical, 12)
                        .background(theme.rose, in: Capsule())
                }
                .buttonStyle(.plain)

                if let existing {
                    Button { onCancel(existing.id); onClose() } label: {
                        Text("取消今次見面")
                            .font(DSText.mono(theme, 12))
                            .foregroundStyle(theme.inkMuted)
                    }
                    .buttonStyle(.plain)
                }
            }
            .padding(20)
            .background(RoundedRectangle(cornerRadius: 24).fill(theme.nav))
            .padding(.horizontal, 20)
        }
    }
}

/// Full-screen prompt shown on the meet-up day until the user submits a selfie.
/// Hosts the camera itself (nested cover) so the caller only presents this one.
struct SelfiePromptView: View {
    @Environment(\.theme) private var theme
    let meetup: Meetup
    let onCapture: (Data) -> Void
    let onLater: () -> Void

    @State private var showCamera = false

    var body: some View {
        ZStack {
            theme.paper.ignoresSafeArea()
            VStack(spacing: 22) {
                Text("( ´ ▽ ` )ﾉ").font(.system(size: 44)).foregroundStyle(theme.rose)
                Text("今日見面喇！🎉")
                    .font(DSText.ui(theme, 22, weight: .semibold))
                    .foregroundStyle(theme.ink)
                Text("影張自拍留念，會自動加入紀念冊")
                    .font(DSText.ui(theme, 14))
                    .foregroundStyle(theme.inkSoft)
                    .multilineTextAlignment(.center)
                    .padding(.horizontal, 36)

                Button { showCamera = true } label: {
                    HStack(spacing: 8) {
                        Image(systemName: "camera.fill")
                        Text("影自拍")
                    }
                    .font(DSText.ui(theme, 16, weight: .semibold))
                    .foregroundStyle(.white)
                    .padding(.horizontal, 28).padding(.vertical, 13)
                    .background(theme.rose, in: Capsule())
                }
                .buttonStyle(.plain)

                Button("遲啲先", action: onLater)
                    .font(DSText.mono(theme, 12))
                    .foregroundStyle(theme.inkMuted)
                    .buttonStyle(.plain)
            }
        }
        .fullScreenCover(isPresented: $showCamera) {
            CameraSheet(
                onPick: { data in showCamera = false; onCapture(data) },
                onCancel: { showCamera = false }
            )
            .ignoresSafeArea()
        }
    }
}

File: android/app/src/main/kotlin/michel/kit/us/features/chat/MeetupViews.kt
package michel.kit.us.features.chat

import androidx.compose.foundation.background
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.outlined.ChevronRight
import androidx.compose.material3.*
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import michel.kit.us.data.Meetup
import michel.kit.us.ui.theme.DSText
import michel.kit.us.ui.theme.LocalLoverColors
import java.time.Instant
import java.time.ZoneId
import java.time.format.DateTimeFormatter

/** Countdown pill at the top of chat. Tapping opens the set dialog. */
@Composable
fun MeetupBanner(upcoming: Meetup?, onClick: () -> Unit) {
    val palette = LocalLoverColors.current
    Row(
        verticalAlignment = Alignment.CenterVertically,
        modifier = Modifier
            .fillMaxWidth()
            .background(palette.roseSoft)
            .clickable(onClick = onClick)
            .padding(horizontal = 14.dp, vertical = 8.dp)
    ) {
        Text("💕", style = DSText.ui(13))
        Spacer(Modifier.width(8.dp))
        if (upcoming != null) {
            val d = upcoming.daysUntil
            Text(
                if (d <= 0) "今日見面！" else "仲有 $d 日見面",
                style = DSText.ui(13, FontWeight.SemiBold).copy(color = palette.ink)
            )
            Spacer(Modifier.width(6.dp))
            Text("· ${upcoming.title}", style = DSText.mono(11).copy(color = palette.inkMuted), maxLines = 1)
        } else {
            Text("設定下次見面", style = DSText.ui(13, FontWeight.SemiBold).copy(color = palette.rose))
        }
        Spacer(Modifier.weight(1f))
        Icon(Icons.Outlined.ChevronRight, contentDescription = null, tint = palette.inkMuted, modifier = Modifier.size(16.dp))
    }
    HorizontalDivider(thickness = 0.5.dp, color = palette.line)
}

/** Date + title picker for the next meet-up. */
@Composable
fun SetMeetupDialog(
    existing: Meetup?,
    onCreate: (dateIso: String, title: String) -> Unit,
    onCancelMeetup: (id: String) -> Unit,
    onClose: () -> Unit
) {
    val palette = LocalLoverColors.current
    val state = rememberDatePickerState()
    var title by remember { mutableStateOf("") }

    Box(
        modifier = Modifier.fillMaxSize().background(Color.Black.copy(alpha = 0.4f)).clickable(onClick = onClose),
        contentAlignment = Alignment.Center
    ) {
        Column(
            horizontalAlignment = Alignment.CenterHorizontally,
            modifier = Modifier.padding(20.dp).clip(RoundedCornerShape(22.dp)).background(palette.nav).padding(16.dp)
                .clickable(interactionSource = remember { androidx.compose.foundation.interaction.MutableInteractionSource() }, indication = null) {}
        ) {
            Text("下次幾時見面？", style = DSText.ui(17, FontWeight.SemiBold).copy(color = palette.ink))
            DatePicker(state = state, title = null, headline = null, showModeToggle = false)
            OutlinedTextField(
                value = title,
                onValueChange = { title = it },
                placeholder = { Text("做咩？(例如：睇戲、食飯)") },
                singleLine = true,
                modifier = Modifier.fillMaxWidth()
            )
            Spacer(Modifier.height(12.dp))
            Button(
                onClick = {
                    val millis = state.selectedDateMillis ?: System.currentTimeMillis()
                    val iso = Instant.ofEpochMilli(millis).atZone(ZoneId.of("Asia/Hong_Kong"))
                        .toLocalDate().format(DateTimeFormatter.ISO_LOCAL_DATE)
                    onCreate(iso, title)
                    onClose()
                },
                colors = ButtonDefaults.buttonColors(containerColor = palette.rose),
                modifier = Modifier.fillMaxWidth()
            ) { Text("確定") }

            existing?.let { m ->
                TextButton(onClick = { onCancelMeetup(m.id); onClose() }) {
                    Text("取消今次見面", style = DSText.mono(12).copy(color = palette.inkMuted))
                }
            }
        }
    }
}

/** Full-screen prompt shown on the meet-up day until the user submits a selfie. */
@Composable
fun SelfiePromptOverlay(onTake: () -> Unit, onLater: () -> Unit) {
    val palette = LocalLoverColors.current
    Box(
        modifier = Modifier.fillMaxSize().background(palette.paper),
        contentAlignment = Alignment.Center
    ) {
        Column(horizontalAlignment = Alignment.CenterHorizontally) {
            Text("( ´ ▽ ` )ﾉ", style = DSText.mono(34).copy(color = palette.rose))
            Spacer(Modifier.height(18.dp))
            Text("今日見面喇！🎉", style = DSText.ui(22, FontWeight.SemiBold).copy(color = palette.ink))
            Spacer(Modifier.height(8.dp))
            Text("影張自拍留念，會自動加入紀念冊", style = DSText.ui(14).copy(color = palette.inkMuted))
            Spacer(Modifier.height(22.dp))
            Button(onClick = onTake, colors = ButtonDefaults.buttonColors(containerColor = palette.rose)) {
                Text("影自拍")
            }
            Spacer(Modifier.height(8.dp))
            TextButton(onClick = onLater) {
                Text("遲啲先", style = DSText.mono(12).copy(color = palette.inkMuted))
            }
        }
    }
}
